cpp/src/gandiva/precompiled/arithmetic_ops.cc

```diff
@@ -158,6 +158,30 @@ NORMALIZE(float32, float32)
 
 #undef NORMALIZE
 
+// floor
+#define FLOOR(IN_TYPE, OUT_TYPE)                                        \
+  FORCE_INLINE                                                          \
+  gdv_##OUT_TYPE floor_##IN_TYPE(gdv_##IN_TYPE in) {                    \
+    return static_cast<gdv_##OUT_TYPE>(std::floor(in));                 \
+  }
+
+FLOOR(float64, int64)
+FLOOR(int64, int64)
+
+#undef FLOOR
+
+// ceil
+#define CEIL(IN_TYPE, OUT_TYPE)                                        \
+  FORCE_INLINE                                                         \
+  gdv_##OUT_TYPE ceil_##IN_TYPE(gdv_##IN_TYPE in) {                    \
+    return static_cast<gdv_##OUT_TYPE>(std::ceil(in));                 \
+  }
+
+CEIL(float64, int64)
+CEIL(int64, int64)
+
+#undef CEIL
```

Comment:
should also add `float32`?
https://github.com/oap-project/arrow/blob/aa061e49c66a760c649b860564170b9735adf121/cpp/src/gandiva/precompiled/extended_math_ops.cc#L81